nifi-extension-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/credentials/service/GCPCredentialsControllerService.java

```diff
@@ -140,6 +141,35 @@ public void migrateProperties(PropertyConfiguration config) {
         config.renameProperty("compute-engine-credentials", USE_COMPUTE_ENGINE_CREDENTIALS.getName());
         config.renameProperty("service-account-json-file", SERVICE_ACCOUNT_JSON_FILE.getName());
         config.renameProperty("service-account-json", SERVICE_ACCOUNT_JSON.getName());
+
+        if (!config.hasProperty(AUTHENTICATION_STRATEGY.getName())) {
```

Comment:
Minor: `hasProperty()` / `isPropertySet()` / `get[Raw]PropertyValue()` methods have signatures with `PropertyDescriptor` parameter so `getName()` could be omitted which is less verbose.